### PR TITLE
Add settlement query

### PR DESCRIPTION
**Pull request by polymath-eric**

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -129,6 +129,37 @@ type Investment @entity {
   datetime: Date!
 }
 
+type PortfolioJson @jsonField {
+  did: String!
+  number: Int!
```
Comment (prashantasdeveloper) [inline schema.graphql:134]: Shouldn't the portfolio number be indexed as well?

File: src/mappings/entities/mapFunding.ts [hunk 2]
```diff
@@ -2,13 +2,14 @@ import { Codec } from "@polkadot/types/types";
...
 import { serializeTicker } from "../util";
+import { EventIdEnum } from "./common";
 
 /**
  * Subscribes to events related to funding events
  */
 export async function mapFunding(
   blockId: number,
-  eventId: string,
+  eventId: EventIdEnum,
```
Comment (prashantasdeveloper) [inline src/mappings/entities/mapFunding.ts:12]: Do change the type of moduleId to `ModuleIdEnum` as well for uniformity

File: src/mappings/entities/mapSettlement.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  EventIdEnum.InstructionExecuted,
+  EventIdEnum.InstructionRejected,
+  EventIdEnum.InstructionFailed,
+];
+
+// Translates events into a settlements table. This inlcudes transfers between a users portfolio combined with completed Instructions.
+export async function mapSettlement(
+  blockId: number,
+  eventId: EventIdEnum,
+  moduleId: string,
```
Comment (prashantasdeveloper) [inline src/mappings/entities/mapSettlement.ts:38]:
```suggestion
  moduleId: ModuleIdEnum,
```

File: src/mappings/entities/mapSettlement.ts [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  const ticker = serializeTicker(params[3]);
+  const amount = getTextValue(params[4]);
+
+  const legs = [
+    {
+      ticker,
+      amount,
+      from: {
+        did: from.did,
+        number: from.kind.user ? from.kind.user : 0,
```
Comment (prashantasdeveloper) [inline src/mappings/entities/mapSettlement.ts:83]: nit pick -> from.kind.user || 0
You can ignore this as well. I prefer the || whenever kind of using it for setting default value

File: tests/integration/instruction.test.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { gql } from "@apollo/client/core";
+import { getApolloClient } from "../util";
+const { query } = getApolloClient();
+
+describe("getInstructions", () => {
```
Comment (prashantasdeveloper) [inline tests/integration/instruction.test.ts:5]:
```suggestion
describe("instructions", () => {
```

File: tests/integration/settlement.test.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
+import { gql } from "@apollo/client/core";
+import { getApolloClient } from "../util";
+const { query } = getApolloClient();
+
+describe("getSettlements", () => {
```
Comment (prashantasdeveloper) [inline tests/integration/settlement.test.ts:5]:
```suggestion
describe("settlements", () => {
```

File: schema.graphql [hunk 7]
```diff
@@ -129,6 +129,37 @@ type Investment @entity {
...
+}
+
+type Leg @jsonField {
+  ticker: String! @index
+  amount: String!
+  from: PortfolioJson
+  to: PortfolioJson
```
Comment (prashantasdeveloper) [inline schema.graphql:141]: there is also a `direction` field in tooling-gql. Are we dropping it?

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

File: schema.graphql (hunk 7 above)
Comment (polymath-eric) [reply schema.graphql:141]: Yeah, tooling-gql would calculate it after querying the DB, but now we don't have any code that happens during the query.

A caller should be able to figure it out with `direction: did === from.did ? 'Outgoing' : 'Incoming'`.

If you have any ideas for keeping I am open to hearing them.

File: schema.graphql [hunk 8]
```diff
@@ -129,6 +129,37 @@ type Investment @entity {
...
+  ticker: String!
+  amount: String!
+  from: PortfolioJson
+  to: PortfolioJson
+}
+
+type Settlement @entity {
+  id: ID!
+  blockId: Int!
+  addresses: [String] @index
```
Comment (raycar5) [inline schema.graphql:147]:
```suggestion
  addresses: [String!]! @index
```

File: schema.graphql [hunk 9]
```diff
@@ -129,6 +129,37 @@ type Investment @entity {
...
+  result: String!
+  legs: [Leg]
```
Comment (raycar5) [inline schema.graphql:149]:
```suggestion
  legs: [Leg!]!
```

**Review by raycar5 — APPROVED** [review]

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**